src/writer/column.rs

```diff
@@ -0,0 +1,225 @@
+use std::marker::PhantomData;
+
+use arrow::{
+    array::{Array, ArrayRef, AsArray},
+    datatypes::{
+        ArrowPrimitiveType, Float32Type, Float64Type, Int16Type, Int32Type, Int64Type, Int8Type,
+        ToByteSlice,
+    },
+};
+use bytes::{Bytes, BytesMut};
+
+use crate::{
+    error::Result,
+    reader::decode::{byte_rle::ByteRleWriter, float::Float, rle_v2::RleWriterV2, SignedEncoding},
```

Comment:
Nit: should we move `byte_rle::ByteRleWriter, rle_v2::RleWriterV2, SignedEncoding` to `writer::encode::*`?

Reply:
I do want to move both the reader & writer versions into a consolidated `encoder` module. Would prefer to keep them together, but will do this as a subsequent step :+1: